File: src/test/java/com/karmanno/plugins/VersionsPluginTest.java
package com.karmanno.plugins;

import nebula.test.PluginProjectSpec;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.BlockJUnit4ClassRunner;

@RunWith(BlockJUnit4ClassRunner.class)
public class VersionsPluginTest extends PluginProjectSpec {
    @Override
    public String getPluginName() {
        return "com.karmanno.plugins.versions";
    }

    @Test
    public void test() {

    }
}

File: src/test/kotlin/com/karmanno/plugins/VersionsPluginTest.kt
package com.karmanno.plugins

import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.io.TempDir
import java.io.File

class VersionsPluginTest {
    @Test
    fun `using plugin id should apply successfully`(@TempDir tempDir: File) {
        // given:
        val project = TestUtils.initializeGradleProject(tempDir)

        // when:
        project.pluginManager.apply("com.karmanno.plugins.semver")

        // then:
        Assertions.assertThat(project.plugins.getPlugin(VersionsPlugin::class.java)).isNotNull
        Assertions.assertThat(project.tasks.findByName("printVersion")).isNotNull
        Assertions.assertThat(project.tasks.findByName("printArtifact")).isNotNull
    }
}